packages/engine/src/interaction/ui/InteractiveModalView.tsx

```diff
@@ -0,0 +1,433 @@
+import React from 'react'
+import { createState } from '@hookstate/core'
+import { createXRUI } from '../../xrui/functions/createXRUI'
+import { useXRUIState } from '@xrengine/engine/src/xrui/functions/useXRUIState'
+import { NavigateNext, NavigateBefore } from '@mui/icons-material'
+
+let callback: any
```

Comment:
We should avoid creating components that rely on global state unless we know for certain we only ever want a single instance active at a time. This callback can be passed in to the `createInteractiveModalView` function, and then passed along as a prop to `InteractiveModalView`. E.g.:

```ts
export function createInteractiveModalView(data: <type this>) {
return createXRUI(() => <InteractiveModalView callback={data.callback}></InteractiveModalView>), createInteractiveModalState(data))
}
```